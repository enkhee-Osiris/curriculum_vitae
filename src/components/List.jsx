import { v4 as uuidv4 } from 'uuid';

function List({ items }) {
  return (
    <ul className="space-y-1 list-disc list-inside">
      {items.map((item) => (
        <li key={uuidv4()}>{item}</li>
      ))}
    </ul>
  );
}

export default List;
