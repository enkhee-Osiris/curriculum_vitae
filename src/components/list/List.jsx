import React from 'react';
import { arrayOf, string } from 'prop-types';

const List = ({ items }) => {
  /* eslint-disable react/no-array-index-key */
  const listItems = items.map((item, index) => (
    <li key={index}>
      {item}
    </li>));

  return <ul>{listItems}</ul>;
};

List.propTypes = {
  items: arrayOf(string).isRequired,
};

export { List as default };
