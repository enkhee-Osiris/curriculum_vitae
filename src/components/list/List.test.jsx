import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

describe('<List />', () => {
  const props = {
    items: ['test'],
  };

  it('should be defined', () => {
    expect(List).toBeDefined();
  });

  it('renders correctly', () => {
    const list = shallow(<List {...props} />);

    expect(list).toMatchSnapshot();
  });

  it('renders li', () => {
    const list = shallow(<List {...props} />);

    expect(list.find('li').length).toBeGreaterThan(0);
  });

  it('renders items', () => {
    const list = shallow(<List {...props} />);

    expect(list.find('li').text()).toEqual('test');
  });
});
