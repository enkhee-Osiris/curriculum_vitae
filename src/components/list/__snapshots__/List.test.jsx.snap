// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    items={
        Array [
            "test",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li>
          test
</li>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            test
</li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
